src/EnumDefinition.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Spatie\Enum;
+
+class EnumDefinition
+{
+    /** @var string|int */
+    public $value;
+
+    public string $label;
+
+    private string $methodName;
+
+    /**
+     * @param string $methodName
+     * @param string|int $value
+     * @param string $label
+     */
+    public function __construct(string $methodName, $value, string $label)
+    {
+        $this->methodName = strtolower($methodName);
+        $this->value = $value;
+        $this->label = $label;
+    }
+
+    public function equals($input): bool
+    {
+        return $this->value === $input
```

Comment:
Right now it's also possible to instantiate an enum by its `value` or after renaming the `label`.
Should we drop this?

Reply:
I think we should drop it, at least I can't come up with cases where this would be beneficial, it only adds complexity in the package, while I think the added "flexibility" adds more confusion than anything else. If there are valid use cases that I can't think of; I'm fine with reconsidering this.

As a side note: the goal of labels is that they allow a textual representation that is allowed to change over time, without having to worry about migrating data. Allowing enums to be constructed from their label seems like partly nullifying that goal.

Reply:
> As a side note: the goal of labels is that they allow a textual representation that is allowed to change over time, without having to worry about migrating data. Allowing enums to be constructed from their label seems like partly nullifying that goal.

:+1: